pkg/controller/kabaneroplatform/targetnamespaces.go

```diff
@@ -0,0 +1,130 @@
+package kabaneroplatform
+import (
+	"context"
+
+	kabanerov1alpha2 "github.com/kabanero-io/kabanero-operator/pkg/apis/kabanero/v1alpha2"
+	
+	"github.com/go-logr/logr"
+	
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func reconcileTargetNamespaces(ctx context.Context, k *kabanerov1alpha2.Kabanero, cl client.Client, reqLogger logr.Logger) error {
+
+	// Rolebinding Template
+	ownerIsController := true
+	rolebindingResource := rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "kabanero-pipeline-deploy-rolebinding",
+			OwnerReferences: []metav1.OwnerReference{
+				metav1.OwnerReference{
+					APIVersion: k.TypeMeta.APIVersion,
+					Kind:       k.TypeMeta.Kind,
+					Name:       k.ObjectMeta.Name,
+					UID:        k.ObjectMeta.UID,
+					Controller: &ownerIsController,
+				},
+			},
+		},
+		Subjects: []rbacv1.Subject{
+			rbacv1.Subject{
+				Kind: "User",
+				Name: "kabanero-pipeline",
+				APIGroup: "rbac.authorization.k8s.io",
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind: "ClusterRole",
+			Name: "kabanero-pipeline-deploy-role",
+			APIGroup: "rbac.authorization.k8s.io",
+		},
+	}
+
+
+	// List of Namespaces we want to bind
+	targetnamespaceList := k.Spec.TargetNamespaces
+	
+	// If targetNamespaces is empty, default to binding to kabanero
+	if len(targetnamespaceList) == 0 {
+		targetnamespaceList = append(targetnamespaceList, "kabanero")
+	}
+
+	/* Get all Rolebindings named kabanero-pipeline-deploy-rolebinding

```

Comment:
This is correct - the controller runtime client doesn't want you looking at anything outside your namespace apparently.  I'm sure we're just using it wrong or something.  There are other places where we've used unstructured instead for this very reason.